tachyon/math/polynomials/univariate/univariate_polynomial.h

```diff
@@ -119,13 +123,26 @@ class UnivariatePolynomial final
     return coefficients_.Evaluate(point);
   }
 
+  template <typename Container>
+  constexpr static Field EvaluateVanishingPolyByRootsSerial(
+      const Container& roots, const Field& point) {
+    Field ret = Field::One();
+    for (size_t i = 0; i < roots.size(); ++i) {
+      ret *= (point - roots[i]);
+    }
+    return ret;
+  }
+
   template <typename Container>
   constexpr static Field EvaluateVanishingPolyByRoots(const Container& roots,
                                                       const Field& point) {
-    return std::accumulate(roots.begin(), roots.end(), Field::One(),
-                           [point](Field& acc, const Field& root) {
-                             return acc *= (point - root);
-                           });
+    size_t chunk_size = base::GetNumElementsPerThread(roots);
+    std::vector<Field> products = base::ParallelizeMapByChunkSize(
+        roots, chunk_size, [&point](absl::Span<const Field> chunk) {
+          return EvaluateVanishingPolyByRootsSerial(chunk, point);
+        });
```

Comment:
This is same with below and if you want to add a threshold please add it as a last argument.

```suggestion
    std::vector<Field> products = base::ParallelizeMapByChunkSize(
        roots, [&point](absl::Span<const Field> chunk) {
          return EvaluateVanishingPolyByRootsSerial(chunk, point);
        });
```

Reply:
@chokobole You mean just `base:: ParallelizeMap` right?